pandas/core/generic.py

```diff
@@ -2782,6 +2782,11 @@ class      (index) object 'bird' 'bird' 'mammal' 'mammal'
             speed    (date, animal) int64 350 18 361 15
         """
 
+        if self.ndim > 2:
+            raise NotImplementedError(
+                "to_xarray is not implemented for {type}"
+                .format(type=type(self)))
+
```

Comment:
Is this check even necessary? ``ndim`` is never > 2, now that panel has been deprecated? Maybe replace with ``assert self.ndim <= 2, ...`` if you really want this check.

The same question to the same checks further below.

Reply:
i'd tend to agree that we could just eliminate all of these checks and make the code simpler.

it would then pass the responsibility to anyone subclassing NDFrame

however, i added the simple assert originally in #25567 after https://github.com/pandas-dev/pandas/pull/25567#discussion_r262923217, (incorrectly I think due to a misunderstanding on my part) and then ended up changing it again after further review https://github.com/pandas-dev/pandas/pull/25567#pullrequestreview-214029237. (I was still misunderstanding the requirement I think)  

so my preference would be to delete all these checks (and redo core.indexing as well) @jreback ?